src/core/connect/dbus_connect.cpp

```diff
@@ -103,6 +95,15 @@ void DBusConnect::process()
             log<level::CRIT>("Fail to process DBus handlers",
                              entry("ERROR=%s", ex.what()),
                              entry("DESC=%s", ex.description()));
+            if (errno == -EIO || errno == -EBADMSG)
+            {
+                log<level::CRIT>(
+                    "DBus connection corrupted. No way to continous processing "
+                    "with current context. Do force restart service.",
```

Comment:
To avoid the mentioned case I did `kill(getpid(), SIGTERM)` to guarantee that the systemd service will handle that correctly.
But you had advised to change the kill to exit.

So, should I revert this to the `kill(getpid(), SIGTERM)`?